src/pages/catalog/Catalog.js

```diff
@@ -81,11 +133,36 @@ function Catalog() {
   };
 
   const clickCourseItemHandler = (key) => {
-    let courseString = localStorage.getItem(key);
-    let course = JSON.parse(courseString);
-    setSpecificCourse(course);
+    const allCourses = JSON.parse(
+      localStorage.getItem(JSON.stringify(filterInfo))
+    );
+    setSpecificCourse(allCourses[parseInt(key)]);
   };
 
+  const courseCardsContent =
+    filtedCourses.length === 0 ? (
```

Comment:
Please also change `filtedCourses.length === 0 ? ` to `filteredCourses == undefined || filteredCourses.length === 0 ? ` so that there won't be stack collapse when we change quarters. 
Note when debugging using in a component using localStorage, always type `localStorage.clear()` in chrome console after making adjustment to the code